feat: add terminal data write event and related functionality

NO-MAP opened the pull request:
#### What It Does
Adds support for the VS Code proposed API `TerminalDataWriteEvent` in Theia.

#### How to Test

**Before the change:**
1. Install the `ms-python` extension in Theia.
2. Run a Python script and observe the following error in the output panel:
```txt
2025-11-17T08:13:03.359Z [Error] Unable to check if shell integration is active TypeError: this.appShell.onDidWriteTerminalData is not a function
    at new m (D:\project\theia\theia-ide\plugins\ms-python-insiders\extension\out\client\extension.js:2:914751)
```

**After the change:**
The error no longer appears, and the feature works as expected.

#### Follow-ups


#### Breaking changes

- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Attribution


#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

ndoschek reviewed (changes requested):
Thank you for the contribution, @NO-MAP!  
I had a first look at the code and added an inline comment, mainly about the structure of the API definition.
Also, do you happen to know an extension that already uses this API, or how did you test it? Otherwise, I can prepare a small test extension to validate the behavior.

Thanks again!

NO-MAP commented:
> Thank you for the contribution, @NO-MAP! I had a first look at the code and added an inline comment, mainly about the structure of the API definition. Also, do you happen to know an extension that already uses this API, or how did you test it? Otherwise, I can prepare a small test extension to validate the behavior.
> 
> Thanks again!

Apologies for the inconvenience caused. I wasn't aware that this was a proposed API.

I've moved the code to the `theia.proposed.terminalDataWriteEvent.d.ts` file and have also updated the PR description accordingly.

I encountered this issue while installing the ms-python extension in Theia. To be clear, this problem doesn't actually prevent ms-python from functioning - it only produces an error message.

Adding this API was actually part of my troubleshooting process for addressing another issue. Since I had already made the changes, I thought it would be helpful to submit them as a PR.

Related issue: https://github.com/eclipse-theia/theia/issues/16570

ndoschek commented on package-lock.json:
This change shouldn’t be needed. Could you please rebase your branch or revert this change? TIA

NO-MAP commented:
@ndoschek It seems I made some incorrect operations that caused the entire commit history to become messy. I will close this PR and create a new one following the correct process. My apologies for the inconvenience.
